### ISSUE TITLE
ComponentProcessingStep was unable to process '*Application_HiltComponents.SingletonC'

### ISSUE THREAD

**Issue by Goooler** [google/dagger#3257]
Build failed from version 2.41, can reproduce this issue on [this branch](https://github.com/Goooler/DemoApp/tree/hilt).

```log
> Task :app:hiltJavaCompileDailyDebug FAILED
error: ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.SingletonC' because 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' could not be resolved.
  
  Dependency trace:
      => element (CLASS): androidx.hilt.lifecycle.ViewModelFactoryModules.ActivityModule
      => element (METHOD): provideFactory(android.app.Activity,android.app.Application,java.util.Map<java.lang.String,javax.inject.Provider<androidx.hilt.lifecycle.ViewModelAssistedFactory<? extends androidx.lifecycle.ViewModel>>>)
      => annotation: @dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory
      => type (ERROR annotation type): dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory
  
  If type 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' is a generated type, check above for compilation errors that may have prevented the type from being generated. Otherwise, ensure that type 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' is on your classpath.
error: ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.ActivityRetainedC' because 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' could not be resolved.
  
  Dependency trace:
      => element (CLASS): androidx.hilt.lifecycle.ViewModelFactoryModules.ActivityModule
      => element (METHOD): provideFactory(android.app.Activity,android.app.Application,java.util.Map<java.lang.String,javax.inject.Provider<androidx.hilt.lifecycle.ViewModelAssistedFactory<? extends androidx.lifecycle.ViewModel>>>)
      => annotation: @dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory
      => type (ERROR annotation type): dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory
  
  If type 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' is a generated type, check above for compilation errors that may have prevented the type from being generated. Otherwise, ensure that type 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' is on your classpath.
error: ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.ActivityC' because 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' could not be resolved.
ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.SingletonC' because 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' could not be resolved.

  
  Dependency trace:
      => element (CLASS): androidx.hilt.lifecycle.ViewModelFactoryModules.ActivityModule
      => element (METHOD): provideFactory(android.app.Activity,android.app.Application,java.util.Map<java.lang.String,javax.inject.Provider<androidx.hilt.lifecycle.ViewModelAssistedFactory<? extends androidx.lifecycle.ViewModel>>>)
      => annotation: @dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory
      => type (ERROR annotation type): dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory
  
  If type 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' is a generated type, check above for compilation errors that may have prevented the type from being generated. Otherwise, ensure that type 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' is on your classpath.
error: ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.FragmentC' because 'dagger.hilt.android.internal.lifecycle.DefaultFragmentViewModelFactory' could not be resolved.
  
  Dependency trace:
      => element (CLASS): androidx.hilt.lifecycle.ViewModelFactoryModules.FragmentModule
      => element (METHOD): provideFactory(androidx.fragment.app.Fragment,android.app.Application,java.util.Map<java.lang.String,javax.inject.Provider<androidx.hilt.lifecycle.ViewModelAssistedFactory<? extends androidx.lifecycle.ViewModel>>>)
      => annotation: @dagger.hilt.android.internal.lifecycle.DefaultFragmentViewModelFactory
      => type (ERROR annotation type): dagger.hilt.android.internal.lifecycle.DefaultFragmentViewModelFactory
  
  If type 'dagger.hilt.android.internal.lifecycle.DefaultFragmentViewModelFactory' is a generated type, check above for compilation errors that may have prevented the type from being generated. Otherwise, ensure that type 'dagger.hilt.android.internal.lifecycle.DefaultFragmentViewModelFactory' is on your classpath.
4 errors

0 problems were found storing the configuration cache.

See the complete report at file:///Users/snebula/StudioProjects/DemoApp/build/reports/configuration-cache/popabyj1ftkzyogj54f7jaoh/bmgev1gjstvd3t2dbnqaodwzu/configuration-cache-report.html

ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.ActivityRetainedC' because 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' could not be resolved.

ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.ActivityC' because 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' could not be resolved.

ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.FragmentC' because 'dagger.hilt.android.internal.lifecycle.DefaultFragmentViewModelFactory' could not be resolved.


```

**Comment by bcorso** [comment]
Hmm, the `androidx.hilt:hilt-lifecycle-viewmodel` artifacts were deprecated in the Dagger [2.34 release](https://github.com/google/dagger/releases/tag/dagger-2.34) in favor of native Hilt API. The missing `DefaultActivityViewModelFactory` class is no longer in the Hilt codebase.

You should be able to fix this using the instructions in the [2.34 release notes](https://github.com/google/dagger/releases/tag/dagger-2.34) to upgrade to the new HiltViewModel API.

**Comment by Goooler** [comment]
Thanks, it works now. May you can mention this behavior change in the 2.41 release note?

**Comment by bcorso** [comment]
Thanks for checking. I've added a link to this discussion in the release notes.

**Comment by noman3qau** [comment]
Removing the dependency on hilt-lifecycle-viewmodel causes the error to go away as it is no longer required in newer versions of hilt. Simply delete this line from your app level build.gradle file if you have it.

implementation  'androidx.hilt:hilt-lifecycle-viewmodel:x.x.x'

**Comment by Muchori** [comment]
Had the same error, this helped.

**Comment by Elmahdaoui59** [comment]
helped

**Comment by wagarcdev** [comment]
on app gradle I had only:

plugins {        
    id 'com.google.dagger.hilt.android'
}

so I added:

plugins {
    id 'com.google.dagger.hilt.android'
    id 'dagger.hilt.android.plugin'
}

and worked fine!

**Comment by motumikat** [comment]
removing  the dependency on hilt-lifecycle-viewmodel causes the error to go away so this one helped me a lot as i was facing the same issues it took me two days to solve the problem

**Comment by RohanTemgire** [comment]
had the same error. helped alot thanks a ton !!!

**Comment by oolyvi** [comment]
> Build failed from version 2.41, can reproduce this issue on [Goooler/DemoApp#31](https://github.com/Goooler/DemoApp/pull/31).
> 
> ```
> > Task :app:hiltJavaCompileDailyDebug FAILED
> error: ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.SingletonC' because 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' could not be resolved.
>   
>   Dependency trace:
>       => element (CLASS): androidx.hilt.lifecycle.ViewModelFactoryModules.ActivityModule
>       => element (METHOD): provideFactory(android.app.Activity,android.app.Application,java.util.Map<java.lang.String,javax.inject.Provider<androidx.hilt.lifecycle.ViewModelAssistedFactory<? extends androidx.lifecycle.ViewModel>>>)
>       => annotation: @dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory
>       => type (ERROR annotation type): dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory
>   
>   If type 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' is a generated type, check above for compilation errors that may have prevented the type from being generated. Otherwise, ensure that type 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' is on your classpath.
> error: ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.ActivityRetainedC' because 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' could not be resolved.
>   
>   Dependency trace:
>       => element (CLASS): androidx.hilt.lifecycle.ViewModelFactoryModules.ActivityModule
>       => element (METHOD): provideFactory(android.app.Activity,android.app.Application,java.util.Map<java.lang.String,javax.inject.Provider<androidx.hilt.lifecycle.ViewModelAssistedFactory<? extends androidx.lifecycle.ViewModel>>>)
>       => annotation: @dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory
>       => type (ERROR annotation type): dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory
>   
>   If type 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' is a generated type, check above for compilation errors that may have prevented the type from being generated. Otherwise, ensure that type 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' is on your classpath.
> error: ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.ActivityC' because 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' could not be resolved.
> ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.SingletonC' because 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' could not be resolved.
> 
>   
>   Dependency trace:
>       => element (CLASS): androidx.hilt.lifecycle.ViewModelFactoryModules.ActivityModule
>       => element (METHOD): provideFactory(android.app.Activity,android.app.Application,java.util.Map<java.lang.String,javax.inject.Provider<androidx.hilt.lifecycle.ViewModelAssistedFactory<? extends androidx.lifecycle.ViewModel>>>)
>       => annotation: @dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory
>       => type (ERROR annotation type): dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory
>   
>   If type 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' is a generated type, check above for compilation errors that may have prevented the type from being generated. Otherwise, ensure that type 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' is on your classpath.
> error: ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.FragmentC' because 'dagger.hilt.android.internal.lifecycle.DefaultFragmentViewModelFactory' could not be resolved.
>   
>   Dependency trace:
>       => element (CLASS): androidx.hilt.lifecycle.ViewModelFactoryModules.FragmentModule
>       => element (METHOD): provideFactory(androidx.fragment.app.Fragment,android.app.Application,java.util.Map<java.lang.String,javax.inject.Provider<androidx.hilt.lifecycle.ViewModelAssistedFactory<? extends androidx.lifecycle.ViewModel>>>)
>       => annotation: @dagger.hilt.android.internal.lifecycle.DefaultFragmentViewModelFactory
>       => type (ERROR annotation type): dagger.hilt.android.internal.lifecycle.DefaultFragmentViewModelFactory
>   
>   If type 'dagger.hilt.android.internal.lifecycle.DefaultFragmentViewModelFactory' is a generated type, check above for compilation errors that may have prevented the type from being generated. Otherwise, ensure that type 'dagger.hilt.android.internal.lifecycle.DefaultFragmentViewModelFactory' is on your classpath.
> 4 errors
> 
> 0 problems were found storing the configuration cache.
> 
> See the complete report at file:///Users/snebula/StudioProjects/DemoApp/build/reports/configuration-cache/popabyj1ftkzyogj54f7jaoh/bmgev1gjstvd3t2dbnqaodwzu/configuration-cache-report.html
> 
> ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.ActivityRetainedC' because 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' could not be resolved.
> 
> ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.ActivityC' because 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' could not be resolved.
> 
> ComponentProcessingStep was unable to process 'io.goooler.demoapp.DemoApplication_HiltComponents.FragmentC' because 'dagger.hilt.android.internal.lifecycle.DefaultFragmentViewModelFactory' could not be resolved.
> ```

Thanks, it works for me on the version 2.44 of Dagger Hilt

**Comment by hey-agrawal** [comment]
> Removing the dependency on hilt-lifecycle-viewmodel causes the error to go away as it is no longer required in newer versions of hilt. Simply delete this line from your app level build.gradle file if you have it.
> 
> implementation 'androidx.hilt:hilt-lifecycle-viewmodel:x.x.x'

thanks its working

**Comment by loay-al-shobaki** [comment]
ComponentProcessingStep was unable to process 'com.example.daggerhilt.MyApp_HiltComponents.SingletonC' because 'dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory' could not be resolved.   i need resolve

**Comment by varshabhatia007** [comment]
> androidx.hilt:hilt-lifecycle-viewmodel

https://github.com/google/dagger/issues/3257#issuecomment-1655179511

**Comment by varshabhatia007** [comment]
> Removing the dependency on hilt-lifecycle-viewmodel causes the error to go away as it is no longer required in newer versions of hilt. Simply delete this line from your app level build.gradle file if you have it.
> 
> implementation 'androidx.hilt:hilt-lifecycle-viewmodel:x.x.x'

Thanks, it is working for me as well without updating any code changes.

**Comment by mr-lonewolfer** [comment]
> Removing the dependency on hilt-lifecycle-viewmodel causes the error to go away as it is no longer required in newer versions of hilt. Simply delete this line from your app level build.gradle file if you have it.
> 
> implementation 'androidx.hilt:hilt-lifecycle-viewmodel:x.x.x'

With this ViewModel init method  even though we properly attached Viewmodel with activity. DO you have any solution for this?

**Comment by sahil1918912** [comment]
> Hmm, the `androidx.hilt:hilt-lifecycle-viewmodel` artifacts were deprecated in the Dagger [2.34 release](https://github.com/google/dagger/releases/tag/dagger-2.34) in favor of native Hilt API. The missing `DefaultActivityViewModelFactory` class is no longer in the Hilt codebase.
> 
> You should be able to fix this using the instructions in the [2.34 release notes](https://github.com/google/dagger/releases/tag/dagger-2.34) to upgrade to the new `HiltViewModel` API.

Thanks bro , it really helps me..

**Comment by k1zlt** [comment]
Thank yooooouuuuu)))))
